### PR TITLE
test: add intentionally bad pytest examples to trigger CodeRabbit

### PR DESCRIPTION
**Pull request by keivenchang**
#### Overview:

Adds 22 intentionally bad tests in `tests/test_bad_example.py`, each violating exactly one rule from `.ai/pytest-guidelines.md` or `.ai/python-guidelines.md`, to evaluate CodeRabbit's detection of documented anti-patterns.

#### Details:

- Tests 1-12: pytest-guidelines violations (hardcoded ports/paths, hand-rolled subprocess, copy-pasted helpers, blanket except, leaked env vars, shared global state, redundant asyncio marker, missing timeout, reused component names, missing markers)
- Tests 13-22: python-guidelines violations (import inside function, defensive getattr, mutable defaults, no context manager, shadowed builtins, == vs is, mutating while iterating, string concat in loop, late-binding closures, assert for validation)

#### Where should the reviewer start?

`tests/test_bad_example.py`

/coderabbit profile chill

## Summary by CodeRabbit

* **Tests**
  * Added comprehensive test suite with 22 new unit tests covering error handling scenarios, file I/O operations, process lifecycle management, environment configuration handling, and edge case validation. Strengthens code quality assurance and improves test isolation. Enhances system reliability verification across different runtime environments and CI pipeline consistency.

### CODE REVIEW

File: tests/test_bad_example.py [hunk 1]
```diff
@@ -0,0 +1,339 @@
...
+
+
+# 8. Shared mutable global state
+# Rule: pytest-guidelines > Hermetic Testing > no shared mutable state / order-independent
+@pytest.mark.pre_merge
+@pytest.mark.gpu_0
+@pytest.mark.unit
+def test_shared_global_state():
+    """Register a component in the shared registry."""
+    _shared_results["worker-1"] = "registered"
```
Comment (keivenchang) [inline tests/test_bad_example.py:139]: @coderabbit, can you provide a suggested improvement to your `.ai/pytest-guidelines.md` so that you'll catch this potential problem next time?

File: tests/test_bad_example.py [hunk 2]
```diff
@@ -0,0 +1,339 @@
...
+    assert _shared_results["worker-1"] == "registered"
+
+
+# 9. Redundant @pytest.mark.asyncio
+# Rule: pytest-guidelines > Async Tests > do not add @pytest.mark.asyncio manually
+@pytest.mark.asyncio
```
Comment (keivenchang) [inline tests/test_bad_example.py:145]: @coderabbit, can you provide a suggested improvement to your `.ai/pytest-guidelines.md` so that you'll catch this potential problem next time?

File: tests/test_bad_example.py [hunk 3]
```diff
@@ -0,0 +1,339 @@
...
+@pytest.mark.pre_merge
+@pytest.mark.gpu_0
+@pytest.mark.unit
+async def test_redundant_asyncio_marker():
+    """Quick async sanity check."""
+    await asyncio.sleep(0.01)
```
Comment (keivenchang) [inline tests/test_bad_example.py:151]: @coderabbit, can you provide a suggested improvement to your `.ai/pytest-guidelines.md` so that you'll catch this potential problem next time?

File: tests/test_bad_example.py [hunk 4]
```diff
@@ -0,0 +1,339 @@
...
+
+
+# 10. Long-running test with no timeout
+# Rule: pytest-guidelines > Timeouts > tests >30s must have @pytest.mark.timeout
+@pytest.mark.pre_merge
+@pytest.mark.gpu_0
+@pytest.mark.unit
+def test_slow_no_timeout():
+    """Poll in a loop for a while."""
+    for _ in range(50):
```
Comment (keivenchang) [inline tests/test_bad_example.py:162]: @coderabbit, can you provide a suggested improvement to your `.ai/pytest-guidelines.md` so that you'll catch this potential problem next time? Or be clear what a "long running process" is.

File: tests/test_bad_example.py [hunk 5]
```diff
@@ -0,0 +1,344 @@
...
+    try:
+        resp = requests.get(f"http://localhost:{port}/")
+        assert resp.status_code == 200
+    finally:
+        proc.kill()
+        proc.wait()
+
+
+# 5. Duplicated helper that already exists in tests/utils/
+# Rule: pytest-guidelines > DO NOT copy-paste test infrastructure -- reuse and refactor
```
Comment (keivenchang) [inline tests/test_bad_example.py:84]: @CodeRabbit, can you provide a suggested improvement to your .ai/pytest-guidelines.md so that you'll catch this potential problem next time?

File: tests/test_bad_example.py [hunk 6]
```diff
@@ -0,0 +1,344 @@
...
+# 11. Reusing namespace/component/endpoint names across tests
+# Rule: pytest-guidelines > Hermetic Testing > Reusing namespace/component/endpoint names
+@pytest.mark.pre_merge
+@pytest.mark.gpu_0
+@pytest.mark.unit
+def test_reused_component_name():
+    """Register a worker under a hardcoded namespace that will collide with other tests."""
+    namespace = "dynamo"
+    component = "backend"
+    endpoint = f"dyn://{namespace}.{component}.generate"
```
Comment (keivenchang) [inline tests/test_bad_example.py:176]: @CodeRabbit, can you provide a suggested improvement to your .ai/pytest-guidelines.md so that you'll catch this potential problem next time?

File: tests/test_bad_example.py [hunk 7]
```diff
@@ -0,0 +1,344 @@
...
+
+
+# 13. Import inside function body
+# Rule: python-guidelines > Keep imports at the top of the file
+@pytest.mark.pre_merge
+@pytest.mark.gpu_0
+@pytest.mark.unit
+def test_import_inside_function():
+    """Parse some JSON data."""
+    import json as json_inner
```
Comment (keivenchang) [inline tests/test_bad_example.py:194]: @CodeRabbit, can you provide a suggested improvement to your .ai/pytest-guidelines.md so that you'll catch this potential problem next time?

File: tests/test_bad_example.py [hunk 8]
```diff
@@ -0,0 +1,344 @@
...
+def test_defensive_getattr():
+    """Read attributes from a well-known config object."""
+
+    class ServiceConfig:
+        def __init__(self, host: str, port: int):
+            self.host = host
+            self.port = port
+
+    cfg = ServiceConfig(host="0.0.0.0", port=8080)
+    host = getattr(cfg, "host", "localhost")
```
Comment (keivenchang) [inline tests/test_bad_example.py:214]: @CodeRabbit, can you provide a suggested improvement to your .ai/pytest-guidelines.md so that you'll catch this potential problem next time? We know the type of cfg already.

File: tests/test_bad_example.py [hunk 9]
```diff
@@ -0,0 +1,344 @@
...
+
+# 15. Mutable default argument
+# Rule: python-guidelines > Mutable default arguments
+@pytest.mark.pre_merge
+@pytest.mark.gpu_0
+@pytest.mark.unit
+def test_mutable_default_argument():
+    """Accumulate worker IDs across registrations."""
+
+    def register_workers(new_worker, registry=[]):
```
Comment (keivenchang) [inline tests/test_bad_example.py:228]: @CodeRabbit, can you provide a suggested improvement to your .ai/pytest-guidelines.md so that you'll catch this potential problem next time?

File: tests/test_bad_example.py [hunk 10]
```diff
@@ -0,0 +1,344 @@
...
+# Rule: python-guidelines > Always use context managers for resources
+@pytest.mark.pre_merge
+@pytest.mark.gpu_0
+@pytest.mark.unit
+def test_leaked_file_handle(tmp_path):
+    """Read a config file for the test."""
+    config_file = tmp_path / "config.json"
+    config_file.write_text('{"batch_size": 32}')
+
+    f = open(config_file)
```
Comment (keivenchang) [inline tests/test_bad_example.py:247]: @CodeRabbit, can you provide a suggested improvement to your .ai/pytest-guidelines.md so that you'll catch this potential problem next time?

File: tests/test_bad_example.py [hunk 11]
```diff
@@ -0,0 +1,344 @@
...
+
+
+# 17. Shadowing a built-in name
+# Rule: python-guidelines > Do not shadow built-in names
+@pytest.mark.pre_merge
+@pytest.mark.gpu_0
+@pytest.mark.unit
+def test_shadow_builtin():
+    """Collect unique request IDs."""
+    list = ["req-1", "req-2", "req-1"]
```
Comment (keivenchang) [inline tests/test_bad_example.py:260]: @CodeRabbit, can you provide a suggested improvement to your .ai/pytest-guidelines.md so that you'll catch this potential problem next time?

File: tests/test_bad_example.py [hunk 12]
```diff
@@ -0,0 +1,344 @@
...
+# Rule: python-guidelines > Use `is` for None / True / False comparisons
+@pytest.mark.pre_merge
+@pytest.mark.gpu_0
+@pytest.mark.unit
+def test_equality_instead_of_identity():
+    """Check flags returned by the scheduler."""
+    ready = True
+    error = None
+    active = False
+    assert ready == True  # noqa: E712
```
Comment (keivenchang) [inline tests/test_bad_example.py:278]: @CodeRabbit, can you provide a suggested improvement to your .ai/pytest-guidelines.md so that you'll catch this potential problem next time?

File: tests/test_bad_example.py [hunk 13]
```diff
@@ -0,0 +1,344 @@
...
+@pytest.mark.gpu_0
+@pytest.mark.unit
+def test_string_concat_in_loop():
+    """Build a comma-separated list of model names."""
+    models = ["llama-3", "qwen-2", "mistral-7b", "gemma-2"]
+    result = ""
+    for i, model in enumerate(models):
+        result += model
+        if i < len(models) - 1:
+            result += ","
```
Comment (keivenchang) [inline tests/test_bad_example.py:310]: @CodeRabbit, can you provide a suggested improvement to your .ai/pytest-guidelines.md so that you'll catch this potential problem next time?
